processor/src/main/java/org/mapstruct/ap/internal/model/source/selector/XmlElementDeclSelector.java

```diff
@@ -112,4 +119,38 @@ else if ( nameMatches.size() > 0 ) {
             return methods;
         }
     }
+
+    private Pair getMatch(SourceMethod sourceMappingMethod, final String targetPropertyName) {
+        final TypeMirror startingMirror = sourceMappingMethod.getExecutable().getReturnType();
+        TypeMirror currentMirror = startingMirror;
+        TypeElement currentElement = sourceMappingMethod.getResultType().getTypeElement();
+
+        while (currentElement != null) {
+            for (Element enclosed : currentElement.getEnclosedElements()) {
+                if (enclosed.getKind().equals( ElementKind.FIELD ) && targetPropertyName != null
+                        && enclosed.getSimpleName().contentEquals( targetPropertyName )) {
+                    XmlElementRef xmlElementRef = enclosed.getAnnotation( XmlElementRef.class );
```

Comment:
Yes, but it's not needed for the JAXB types. We use the prisms for the MapStruct annotations to avoid having to provide them on the processor path. As JAXB is provided by the JDK, accessing the annotation is fine.